packages/chain-bridge/src/router/BridgeRouter.sol

```diff
@@ -606,6 +775,282 @@ contract BridgeRouter is
+
+            // Apply overrides to the payload
+            effectivePayload = _applyRetryOverrides(
+                r.operationType,
+                effectivePayload,
+                overrides,
+                r.sourceChainId
+            );
+        } else {
+            // Validate original payload even when no overrides are applied
+            _validateRetryPayload(
+                r.operationType,
+                effectivePayload,
+                r.sourceChainId,
+                effectiveAdapter
+            );
+        }
+
+        // Validate that payload decodes and carries the same operationId and sourceChainId
+        (bytes32 decodedId, uint16 sourceChainId) = _decodeOperationMeta(
+            r.operationType,
+            effectivePayload
+        );
+
+        try
+            this._processDelivery(
+                r.operationType,
+                effectivePayload,
+                effectiveAdapter
+            )
+        {
+            _clearFailedDelivery(operationId);
+            emit OperationRetrySucceeded(
+                operationId,
+                r.operationType,
+                effectiveAdapter
+            );
+        } catch (bytes memory err) {
+            // Do not create a new failure record; update existing metadata only
+            FailedDeliveryRecord storage existing = failedDeliveries[
+                operationId
+            ];
+            existing.failedAt = block.timestamp;
+
+            emit OperationRetryFailed(
+                operationId,
+                r.operationType,
+                effectiveAdapter,
+                err
+            );
+        }
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                        RETRY OVERRIDE FUNCTIONS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Applies retry overrides to the operation payload
+     * @param operationType Type of operation being retried
+     * @param originalPayload The original payload
+     * @param overrides The override parameters
+     * @param sourceChainId The source chain ID for validation
+     * @return modifiedPayload The payload with overrides applied
+     */
+    function _applyRetryOverrides(
+        BridgeTypes.OperationType operationType,
+        bytes memory originalPayload,
+        RetryOverrideParams memory overrides,
+        uint16 sourceChainId
+    ) internal view returns (bytes memory modifiedPayload) {
+        if (operationType == BridgeTypes.OperationType.TRANSFER_ASSET) {
+            BridgeTypes.RelayedTransferParams memory params = abi.decode(
+                originalPayload,
+                (BridgeTypes.RelayedTransferParams)
+            );
+
+            // Apply recipient override only
+            // Asset addresses are handled by adapters for cross-chain compatibility
+            if (overrides.recipient != address(0)) {
+                params.recipient = overrides.recipient;
+            }
+
```

Comment:
this check is redundant - since the `receipient` is the only paramater that can be overriden - hence if overrrideData length is > 0 - then it's the receipient